sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -576,57 +563,71 @@ private void OverrideBindingName(INamedTypeSymbol attributeClass, ref string arg
                 }
             }
 
+            private bool HasCardinality(AttributeData attribute)
+            {
+                var attrClass = attribute.AttributeClass;
+                var isBatchedProp = attrClass!.GetMembers().Where(m => string.Equals(m.Name, Constants.FunctionMetadataBindingProps.IsBatchedKey, StringComparison.OrdinalIgnoreCase)).SingleOrDefault();
+
+                if (isBatchedProp != null)
+                {
+                    return true;
+                }
+
+                return false;
+            }
+
             /// <summary>
-            /// This method verifies that an EventHubsTrigger matches our expectations on cardinality (isBatched property). If isBatched is set to true, the parameter with the
+            /// This method verifies that a binding that has Cardinality (isBatched property) is valid. If isBatched is set to true, the parameter with the
             /// attribute must be an enumerable type.
             /// </summary>
-            private bool IsEventHubsTriggerValid(IParameterSymbol parameterSymbol, TypeSyntax? parameterTypeSyntax, SemanticModel model, AttributeData attribute, out DataType dataType, out Cardinality cardinality)
+            private bool IsCardinalityValid(IParameterSymbol parameterSymbol, TypeSyntax? parameterTypeSyntax, SemanticModel model, AttributeData attribute, out DataType dataType)
             {
                 dataType = DataType.Undefined;
-                cardinality = Cardinality.Undefined;
+                var cardinalityIsNamedArg = false;
 
```

Comment:
Note: No longer handling cardinality as a special case. All _Named Arguments_ and _Arguments with default values_ are handled in [this method](https://github.com/Azure/azure-functions-dotnet-worker/blob/0fd7bf6aef005e4b8a14874506bf7a8ad7ad73ef/sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Parser.cs#L445).